src/main/java/com/manywho/services/email/email/decisions/clients/EngineClient.java

```diff
@@ -48,7 +49,18 @@ public String sendResponseToEngine(EmailDecisionRequest emailRequest, String str
                 flowState.sync();
             }
 
-            return runClient.join(emailRequest.getTenantId().toString(), flowState.getState().toString()).execute().body().getJoinFlowUri();
+            /**
+             * The sync() call can potentially clear out the invoke response that hangs off of the flowState
+             * instance (this seems to typically occur when there are no map elements to move on to). If an
+             * invoke response has been generated by the SDK, then grab the state ID from there, if not
+             * then just grab it from the incoming request.
+             * Being overly cautious here, but am unsure as to whether the a state ID might change when invoking
+             * the flows state, for instance with a flow that utilses a child flow (old subflows). So will continue
+             * to get the state ID from flowState if possible.
+             */
```

Comment:
In my understanding is not possible that the state Id change during execution in new or old subflows
@jonjomckay, @jacek-wojciechowski-dell could you please confirm?

Reply:
@jamesbrattBoomi I got confirmation that the state doesn't change. I don't mind being overly cautious but could you please update the comment to be aligned to this.

Reply:
There shouldn't be any case where performing a `SYNC` doesn't return an invoke response - every call to invoke a state will return a response containing the state ID (amongst other things), unless the response returns an error. 

This feels like a bug in the SDK's `FlowClient`, so you should probably fix it there (which fixes it for everything), rather than just patch it in this service.

Reply:
Oki doki, I will check again in the SDK to see what could be causing the invoke response to return null in the sync call